### PR TITLE
Matrix factorization

### PR DESCRIPTION
**Pull request by danich1**
Matrix Factorization section of the knowledge graph review.

### CODE REVIEW

File: content/04.network_applications.md [hunk 1]
```diff
@@ -10,7 +10,31 @@
 
 #### Matrix Factorization
 
-1. Mention techniques for these with some papers
+Matrix factorization is a technique that uses linear algebra to map high dimensional data into a low dimensional space.
+This projection is accomplished by decomposing a matrix into a set of small rectangular matrices.
+In the context of knowledge graphs the input matrix would be an adjacency matrix where the rows and columns are nodes and each entry in this matrix represents the presence of an edge between two nodes.
+The output of this technique would be a small matrix where each row represents a low dimensional embedding for every node in the adjacency matrix.
```
Comment (cgreene) [inline content/04.network_applications.md:19]: Hmm - I usually think of this approach as decomposing a large matrix into two smaller matrices. Do you mean something different than this?

File: content/04.network_applications.md [hunk 2]
```diff
@@ -10,7 +10,31 @@
...
+In this section we discuss selected approaches that utilize the above methods to project knowledge graphs into a low dimensional space.
+
+Laplacian eigenmap is an unsupervised approach that learns the local structure of a knowledge graph.
+Given an adjacency matrix (A) this algorithm first calculates a laplacian matrix.
+A laplacian matrix (L) is a matrix that is calculated by subtracting a knowledge graph's degree matrix (D) (a diagonal matrix where each entry represents the number of edges connected to a node) from its adjacency matrix ($L = D-A$).
+Using the laplacian matrix, this algorithm then calculates eigenvalues and eigenvectors ($Lx = \lambda Dx$).
+The calculated eigenvectors represent the nodes of the knowledge graph [@doi:10.7551/mitpress/1120.003.0080].
+A number of approaches have used variants of this algorithm to perform their own node projection [@raw:gongreplearn; @doi:10.1109/TKDE.2016.2606098; @raw:hanreplearn].
+This type of methodology is easy to execute the only input required is an adjacency matrix and a degree matrix; however, this approach relies on a linear mapping function, which may not be appropriate given certain biomedical problems.
```
Comment (cgreene) [inline content/04.network_applications.md:29]: Are there any ways that this has been used? What does it work well with, and what does it work poorly with?

File: content/04.network_applications.md [hunk 3]
```diff
@@ -10,7 +10,31 @@
...
+
+Following Laplacian eigenmaps, other works constructed variants of the adjacency matrix before undergoing matrix factorization [@doi:10.1145/2806416.2806512; @raw:zhaoreplearn; @raw:yangreplearn; @raw:tureplearn].
```
Comment (cgreene) [inline content/04.network_applications.md:31]: My understanding from this description is that all of these methods use random walk and then do matrix factorization on the matrix after some iterative random walk procedure. Is something like that correct?

File: content/04.network_applications.md [hunk 4]
```diff
@@ -10,7 +10,31 @@
...
+These methods simulate the random walk that Deepwalk uses via matrix multiplications.
+The adjacency matrix is multiplied with a normalized degree matrix to represent transition probabilities.
+This construct matrix turns the process of random walks into simple matrix multiplication.
+Lastly these methods use a form of matrix factorization on to project the nodes into a low dimensional space.
+
+Overall matrix factorization is a powerful technique that uses a matrix similar to an adjacency matrix as input.
```
Comment (cgreene) [inline content/04.network_applications.md:37]: What are the implications of using an adjacency matrix or the results of an iterative random walk procedure? I'm imagining that it makes sense to use one in one case and another in a different case.

File: content/04.network_applications.md [hunk 5]
```diff
@@ -10,7 +10,31 @@
...
+Despite reported success, these methods work only for regular size knowledge graphs.
+If a knowledge graph were to expand to gargantuan size, these type of approaches would become impractical.
+Furthermore, these methods treat all edge types the same, which may not be appropriate for knowledge graphs with heterogeneous edge types.
```
Comment (cgreene) [inline content/04.network_applications.md:40]: Would this necessarily be true if the random walk procedure had some edge-type specificity to it? Is that the way to make these things edge type specific (maybe nobody has done this yet?)?

File: content/04.network_applications.md [hunk 6]
```diff
@@ -10,7 +10,31 @@
...
+
+Following Laplacian eigenmaps, other works constructed variants of the adjacency matrix before undergoing matrix factorization [@doi:10.1145/2806416.2806512; @raw:zhaoreplearn; @raw:yangreplearn; @raw:tureplearn].
+These methods simulate the random walk that Deepwalk uses via matrix multiplications.
+The adjacency matrix is multiplied with a normalized degree matrix to represent transition probabilities.
+This construct matrix turns the process of random walks into simple matrix multiplication.
+Lastly these methods use a form of matrix factorization on to project the nodes into a low dimensional space.
+
+Overall matrix factorization is a powerful technique that uses a matrix similar to an adjacency matrix as input.
+Despite reported success, these methods work only for regular size knowledge graphs.
+If a knowledge graph were to expand to gargantuan size, these type of approaches would become impractical.
```
Comment (cgreene) [inline content/04.network_applications.md:39]: I probably wouldn't use "gargantuan" in a paper. Maybe a better way to approach this is describing how the time and memory complexity scale as a factor of the number of nodes.

**Review by cgreene — COMMENTED** [review]
On this one I have more questions than answers!

**Comment by danich1** [comment]
This PR has gotten really messy thanks to conflicts with the manual-references.json file. Going to close and reopen a new PR with majority of these problems fixed. Plus the content for this section will be updated.

**Pull request closed without merging**